functions/__sf_section_docker.fish

```diff
@@ -0,0 +1,51 @@
+#
+# Docker
+#
+# Current Docker version and Machine name.
+
+function __sf_section_docker -d "Display docker version and machine name"
+    # ------------------------------------------------------------------------------
+    # Configuration
+    # ------------------------------------------------------------------------------
+
+    __sf_util_set_default SPACEFISH_DOCKER_SHOW true
+    __sf_util_set_default SPACEFISH_DOCKER_PREFIX "is "
+    __sf_util_set_default SPACEFISH_DOCKER_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+    __sf_util_set_default SPACEFISH_DOCKER_SYMBOL "🐳 "
+    __sf_util_set_default SPACEFISH_DOCKER_COLOR cyan
+    __sf_util_set_default SPACEFISH_DOCKER_VERBOSE_VERSION false
+
+    # ------------------------------------------------------------------------------
+    # Section
+    # ------------------------------------------------------------------------------
+
+    [ $SPACEFISH_DOCKER_SHOW = false ]; and return
+
+    # Show Docker version only if docker is installed
+    type -q docker; or return
+
+    # Show docker version only when pwd has dockerfile or docker-compose.yml or COMPOSE_FILE
+    [ -f ./Dockerfile -o -f ./docker-compose.yml ]; or test "$COMPOSE_FILE"; or return
+
+    # Show Docker version only if docker deamon is up
+    docker version > /dev/null 2>&1
+    [ $status -ne 0 ]; and return
+
+    set -l docker_version v(docker version -f "{{.Server.Version}}" ^/dev/null)
+    # if docker daemon isn't running you'll get an error like 'Bad response from Docker engine'
+    [ -z docker_version ]; and return
+
+    if test $SPACEFISH_DOCKER_VERBOSE_VERSION = false
```

Comment:
I suggest wrapping test arguments in double-quotes.

There is some room for issues if there are whitespace or symbols in `test` arguments, as is [detailed in the docs](https://fishshell.com/docs/current/commands.html#test). After explaining those issues, it goes on to say:
> Since it is always safe to enclose variables in double-quotes when used as test arguments that is the recommended practice.

```suggestion
    if test "$SPACEFISH_DOCKER_VERBOSE_VERSION" = "false"
```